ts/Series/FlowMap/FlowMapSeries.ts

```diff
@@ -0,0 +1,829 @@
+         * arrays from different sources.
+         *
+         * @default ['from', 'to', 'weight']
+         */
+        keys: ['from', 'to', 'weight'],
+
+        /**
+         * The arrow type and size properties.
+         *
+         * @declare Highcharts.SeriesFlowMapSeriesOptionsObject
+         * @since   next
+         */
+        markerEnd: {
+            /**
+             * Whether a flowmap point should end with a symbol or not.
+             *
+             * @since next
+             * @type  {boolean}
+             */
+            enabled: true,
+            /**
+             * Height of a flowmap point's symbol.
+             *
+             * @since next
+             * @type  {number|string}
+             */
+            height: '40%',
+            /**
+             * Width of a flowmap point's symbol.
+             *
+             * @since next
+             * @type  {number|string}
+             */
+            width: '40%'
+        },
+
+        /**
+         * Maximum flow map links weight. It will automatically
+         * size between the minWeight and maxWeight.
+         *
+         * @since   next
+         */
+        maxWeight: 25,
+
+        /**
+         * Minimum flow map links weight. It will automatically
+         * size between the minWeight and maxWeight.
+         *
+         * @declare Highcharts.SeriesFlowMapSeriesOptionsObject
+         * @since   next
+         */
+        minWeight: 5,
+
+        tooltip: {
+            /**
+             * The HTML of the tooltip header line. Variables are enclosed by
+             * curly brackets. Available variables are point.key, series.name,
+             * series.color and other members from the point and series
+             * objects. The point.key variable contains the category name, x
+             * value or datetime string depending on the type of axis. For
+             * datetime axes, the point.key date format can be set using
+             * tooltip.xDateFormat.
+             *
+             * TODO: check why it's not generated in API
+             */
+            headerFormat: '<span style="font-size: 10px">{series.name}</span><br/>',
+            pointFormat: '{point.options.from} \u2192 {point.options.to}: <b>{point.weight}</b><br/>'
+        },
+
+        /**
+         * The weight of the link determines how thick it will be compared to
+         * other weights.
+         *
+         * @type    {number}
+         * @since   next
+         */
+        weight: void 0
+
+    } as FlowMapSeriesOptions);
+
+    /* *
+     *
+     *  Static Function
+     *
+     * */
+
+    /**
+     * Get vector length.
+     * @private
+     */
+    public static getLength(x: number, y: number): number {
+        return Math.sqrt(x * x + y * y);
+    }
+
+    /**
+     * Return a normalized vector.
+     * @private
+     */
+    public static normalize(x: number, y: number): Array<number> {
+        const length = FlowMapSeries.getLength(x, y);
```

Comment:
Why not the simpler `this.getLength`? Doesn't look like it needs to be called on the class.